src/dune_rules/dialect.ml

```diff
@@ -253,17 +275,24 @@ module DB = struct
         else { Ml_kind.Dict.impl; intf } :: s)
       |> List.sort ~compare:(Ml_kind.Dict.compare (Option.compare String.compare))
     in
-    t.extensions_for_merlin <- Some v;
-    v
+    let readers =
+      String.Map.fold by_name ~init:String.Map.empty ~f:(fun d s ->
+        let add kind s =
+          match merlin_reader d kind with
+          | None -> s
+          | Some (extension, reader) ->
+            (* Ok to use [add_exn] here as we've validating below that we have
```

Comment:
Do you mean `above` here instead of `below`?

Reply:
The code which validates extension is registered once is [below](https://github.com/ocaml/dune/pull/8567/files/9669feffd1a3b66e27a7783e5a2c6ed41cd61f8b#diff-e77f32927aca7df771b4e00ecc3315a416133fdc68e1c9133139a60ef399fec1R310). I'll add a more specific reference though, good catch.